ocaml/xapi/xapi_vbd_helpers.ml

```diff
@@ -169,12 +169,21 @@ let valid_operations ~expensive_sharing_checks ~__context record _ref' : table =
     if not vdi_record.Db_actions.vDI_managed
     then set_errors Api_errors.vdi_not_managed [ _ref ] all_ops;
 
-    let vdi_operations_besides_copy =
+    let vdi_operations_cannot_be_performed_live =
+      let illegal_operation = function
+        | `snapshot
+        | `resize_online
+        | `blocked
+        | `clone
+        | `mirror
+        | `copy -> false
+        | _     -> true
+      in
```

Comment:
Really appreciate your suggestion, I have concern about this since my lack of knowledge about the SM module. If there are retry mechanism about the operation, I think we only should consider the valid long-term operations here, like `copy`, `mirror` and `clone`. When I try to verify the `clone` operation, I realize that `clone` only can be performed when the VM status is `halted`, so we needn't consider `clone` here. I will update the source code to check whether the operation is `copy` or `mirror`.